internal/core/server/workspace.go

```diff
@@ -151,6 +151,19 @@ func (ws *Workspace) Apply(ctx context.Context, input *api.ApplyInput) (*api.App
 	}
 	manifestComponents := componentSet.Components
 
+	manifestComponentDeps := make([][]string, len(manifestComponents))
+	for i, c := range manifestComponents {
+		deps, err := ws.getComponentDeps(ctx, api.ComponentDescription{
+			Name: c.Name,
+			Type: c.Type,
+			Spec: c.Spec,
+		})
+		if err != nil {
+			ws.logEventf(ctx, "error getting %q deps: %v", c.Name, err)
```

Comment:
Why isn't this considered a fatal error?

Reply:
I'll add a comment. I think that if it were a fatal error, then it would be impossible to stop components that have broken provider implementations or partially invalid specs that fail dependency analysis.